### PR TITLE
Update Committee Helpers to v0.6.0

### PR DESCRIPTION
**Pull request by 0xKiwi**
Part of #2307 

Changes made: 
Updated EpochCommitteeCount to latest spec
Updated CrosslinkCommitteesAtSlot to latest spec
Added ComputeCommittee
Added ShardDelta

### CODE REVIEW

**Comment by terencechain** [comment]
@0xKiwi can we address the following?
```
beacon-chain/core/helpers/committee.go:558:6: `prevEpochCommitteesAtSlot` is unused (deadcode)
func prevEpochCommitteesAtSlot(state *pb.BeaconState, slot uint64) ([]*CrosslinkCommittee, error) {
     ^
beacon-chain/core/helpers/committee.go:587:6: `currEpochCommitteesAtSlot` is unused (deadcode)
func currEpochCommitteesAtSlot(state *pb.BeaconState, slot uint64) ([]*CrosslinkCommittee, error) {
     ^
beacon-chain/core/helpers/committee.go:631:6: `nextEpochCommitteesAtSlot` is unused (deadcode)
func nextEpochCommitteesAtSlot(state *pb.BeaconState, slot uint64, registryChange bool) ([]*CrosslinkCommittee, error) {
     ^
beacon-chain/core/helpers/committee.go:29:6: U1000: type `shufflingInput` is unused (unused)
type shufflingInput struct {
     ^
beacon-chain/core/helpers/committee.go:701:6: U1000: func `crosslinkCommittees` is unused (unused)
func crosslinkCommittees(state *pb.BeaconState, input *shufflingInput) ([]*CrosslinkCommittee, error) {
```

File: beacon-chain/core/balances/rewards_penalties.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func ExpectedFFGSource(
 				justifiedAttestingBalance /
 				totalBalance
 	}
-	activeValidatorIndices := helpers.ActiveValidatorIndices(state.ValidatorRegistry, helpers.CurrentEpoch(state))
```
Comment (nisdas) [inline beacon-chain/core/balances/rewards_penalties.go]: please revert all these changes, we use the validator registry in `ActiveValidatorIndices`, this PR has reverted it

File: beacon-chain/core/validators/validator.go [hunk 2]
```diff
@@ -167,10 +167,10 @@ func ActivateValidator(state *pb.BeaconState, idx uint64, genesis bool) (*pb.Bea
...
 	state.ValidatorRegistry[idx] = validator
 
-	log.WithFields(logrus.Fields{
-		"index":           idx,
-		"activationEpoch": validator.ActivationEpoch - params.BeaconConfig().GenesisEpoch,
-	}).Info("Validator activated")
+	// log.WithFields(logrus.Fields{
+	// 	"index":           idx,
+	// 	"activationEpoch": validator.ActivationEpoch - params.BeaconConfig().GenesisEpoch,
+	// }).Info("Validator activated")
```
Comment (shayzluf) [inline beacon-chain/core/validators/validator.go]: why is it commented?

**Comment by 0xKiwi** [comment]
Ah, nice catch, it would clutter my test logs haha, I'll add it back.

File: beacon-chain/core/helpers/committee.go [hunk 3]
```diff
@@ -27,31 +27,24 @@ type CrosslinkCommittee struct {
...
 //        )
 //    ) * SLOTS_PER_EPOCH
-func EpochCommitteeCount(activeValidatorCount uint64) uint64 {
+func EpochCommitteeCount(beaconState *pb.BeaconState, epoch uint64) uint64 {
 	var minCommitteePerSlot = uint64(1)
-
+	activeValidatorCount := uint64(len(ActiveValidatorIndices(beaconState.ValidatorRegistry, epoch)))
 	// Max committee count per slot will be 0 when shard count is less than epoch length, this
 	// covers the special case to ensure there's always 1 max committee count per slot.
 	var maxCommitteePerSlot = minCommitteePerSlot
```
Comment (shayzluf) [inline beacon-chain/core/helpers/committee.go]:
```suggestion
	var committeePerSlot = minCommitteePerSlot
```

File: beacon-chain/core/helpers/committee.go [hunk 4]
```diff
@@ -121,54 +112,105 @@ func PrevEpochCommitteeCount(state *pb.BeaconState) uint64 {
...
-		return prevEpochCommitteesAtSlot(state, slot)
+		previousShardDelta := ShardDelta(state, prevEpoch)
+		startShard = (state.LatestStartShard - previousShardDelta) % shardCount
 	case nextEpoch:
-		return nextEpochCommitteesAtSlot(state, slot, registryChange)
+		currentShardDelta := ShardDelta(state, currentEpoch)
+		startShard = (state.LatestStartShard + currentShardDelta) % shardCount
 	default:
 		return nil, fmt.Errorf(
 			"input committee epoch %d out of bounds: %d <= epoch <= %d",
```
Comment (shayzluf) [inline beacon-chain/core/helpers/committee.go]: out of bound should have `previous_epoch <= epoch <= next_epoch`

File: beacon-chain/core/helpers/committee_test.go [hunk 5]
```diff
@@ -126,6 +144,35 @@ func TestNextEpochCommitteeCount_OK(t *testing.T) {
...
 }
 
+func TestShardDelta_OK(t *testing.T) {
+	minShardDelta := params.BeaconConfig().ShardCount -
+		params.BeaconConfig().ShardCount/params.BeaconConfig().SlotsPerEpoch
+	tests := []struct {
+		validatorCount uint64
+		shardCount     uint64
+	}{
+		{0, params.BeaconConfig().SlotsPerEpoch},
```
Comment (shayzluf) [inline beacon-chain/core/helpers/committee_test.go]: could you name your assignments? easier to read

**Pull request merged**